src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx

```diff
@@ -52,7 +53,10 @@ export function DataSourceMenu<T>(props: DataSourceMenuProps<T>): ReactElement |
     );
   }
 
-  function renderDataSourceSelectable(config: DataSourceSelectableConfig): ReactElement | null {
+  function renderDataSourceSelectable(
+    config: DataSourceSelectableConfig,
+    ui: IUiSettingsClient
```

Comment:
looks like we already extract the uiSettings props above, and we don't need to add it here
```suggestion
```